codeowners/src/codeowners.rs

```diff
@@ -121,6 +155,58 @@ impl BindingsOwners {
     }
 }
 
+async fn associate_codeowners(
+    codeowners_matchers: Arc<HashMap<String, Option<Owners>>>,
+    bundle_upload_id_and_file_path: BundleUploadIDAndFilePath,
+) -> Vec<String> {
+    let (bundle_upload_id, file) = bundle_upload_id_and_file_path;
+    let codeowners_matcher = codeowners_matchers.get(&bundle_upload_id);
+    match (codeowners_matcher, &file) {
+        (Some(Some(owners)), Some(file)) => match owners {
+            Owners::GitHubOwners(gho) => gho
+                .of(file)
+                .unwrap_or_default()
+                .iter()
+                .map(ToString::to_string)
+                .collect(),
+            Owners::GitLabOwners(glo) => glo
+                .of(file)
+                .unwrap_or_default()
+                .iter()
+                .map(ToString::to_string)
+                .collect(),
+        },
+        _ => Vec::new(),
+    }
+}
+
+pub async fn associate_codeowners_multithreaded(
+    codeowners_matchers: HashMap<String, Option<Owners>>,
+    to_associate: Vec<BundleUploadIDAndFilePath>,
+) -> Result<Vec<Vec<String>>> {
+    let codeowners_matchers = Arc::new(codeowners_matchers);
+    let mut results = vec![None; to_associate.len()];
+    let mut tasks = Vec::new();
+
+    for (i, bundle_upload_id_and_file_path) in to_associate.into_iter().enumerate() {
+        let codeowners_matchers = Arc::clone(&codeowners_matchers);
+        let task = task::spawn(async move {
+            (
+                i,
+                associate_codeowners(codeowners_matchers, bundle_upload_id_and_file_path).await,
+            )
+        });
+        tasks.push(task);
+    }
+
+    for task in tasks {
+        let (i, result) = task.await?;
+        results[i] = Some(result);
+    }
+
+    Ok(results.into_iter().flatten().collect())
+}
```

Comment:
Very similar feedback to the above. `BundleId` isn't related to codeowners, so we shouldn't have any logic related to it here:

```suggestion
pub async fn associate_codeowners_multithreaded<T: AsRef<Path>>(
    to_associate: Vec<(Owners, T)>,
) -> Result<Vec<Vec<String>>> {
    let tasks = to_associate.iter().enumerate().map(|(i, (owners, file))| {
        task::spawn(async move {
            (i, associate_codeowners(&owners, file))
        })
    }).collect::<Vec<_>>();

    let mut results = Vec::with_capacity(task.len());
    for task in tasks {
        let (i, result) = task.await?;
        results[i] = result;
    }

    Ok(results)
}
```